src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.PerAnalyzerState.cs

```diff
@@ -429,7 +444,7 @@ public void OnCompilationEventGenerated(CompilationEvent compilationEvent, Analy
                             actionCounts.HasAnyExecutableCodeActions)
                         {
                             needsAnalysis = true;
-                            _pendingDeclarations[symbol] = _currentlyAnalyzingDeclarationsMapPool.Allocate();
+                            _pendingDeclarations[symbol] = null;
```

Comment:
This is the core change in this source file. We now delay the allocation of per-symbol pending declarations map until the symbol's analysis starts. Rest of the changes in this file handle the null value appropriately.

Reply:
❔ Do we need to insert it here, or can we leverage the `TryGetValue` result being false to double as not yet initialized?

Reply:
We do need to insert it. We actually remove the entry once the declaration has been fully analyzed, so lack of entry already indicates a separate state.